### PR TITLE
Bot Resource for lexv2

### PR DESCRIPTION
**Pull request by nam054**
### Description

Add bot resource to lexv2 service.

### Relations

Closes #0000

### Output from Acceptance Testing

```console
> make testacc TESTS=TestAccLexV2ModelsBot_ PKG=lexv2models
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lexv2models/... -v -count 1 -parallel 20 -run='TestAccLexV2ModelsBot_'  -timeout 180m
=== RUN   TestAccLexV2ModelsBot_basic
=== PAUSE TestAccLexV2ModelsBot_basic
=== RUN   TestAccLexV2ModelsBot_tags
=== PAUSE TestAccLexV2ModelsBot_tags
=== RUN   TestAccLexV2ModelsBot_disappears
=== PAUSE TestAccLexV2ModelsBot_disappears
=== CONT  TestAccLexV2ModelsBot_disappears
=== CONT  TestAccLexV2ModelsBot_basic
=== CONT  TestAccLexV2ModelsBot_tags
--- PASS: TestAccLexV2ModelsBot_disappears (24.47s)
--- PASS: TestAccLexV2ModelsBot_basic (24.88s)
--- PASS: TestAccLexV2ModelsBot_tags (51.44s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/lexv2models        54.486s

...
```

### CODE REVIEW

File: internal/framework/flex/int_test.go [hunk 1]
```diff
@@ -124,3 +124,39 @@ func TestInt64ToFrameworkLegacy(t *testing.T) {
 		})
 	}
 }
+
+func TestInt32ToFramework(t *testing.T) {
```
Comment (YakDriver) [inline internal/framework/flex/int_test.go:128]: 🎉 Unit tests

File: internal/service/lexv2models/bot.go [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+
+func (r *resourceBot) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	r.SetTagsAll(ctx, req, resp)
+}
+
+func (r *resourceBot) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func waitBotCreated(ctx context.Context, conn *lexmodelsv2.Client, id string, timeout time.Duration) (*lexmodelsv2.DescribeBotOutput, error) {
```
Comment (YakDriver) [inline internal/service/lexv2models/bot.go:389]: Nice, with the naming

File: internal/service/lexv2models/bot.go [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+	}
+
+	if out == nil || out.BotId == nil {
+		return nil, tfresource.NewEmptyResultError(in)
+	}
+
+	return out, nil
+}
+
+func flattenDataPrivacy(apiObject *awstypes.DataPrivacy) (types.List, diag.Diagnostics) {
```
Comment (YakDriver) [inline internal/service/lexv2models/bot.go:481]: I haven't used the autoflex stuff yet but we should try it for some of these resources.

File: internal/service/lexv2models/bot_test.go [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func TestAccLexV2ModelsBot_basic(t *testing.T) {
+	ctx := acctest.Context(t)
+	// TIP: This is a long-running test guard for tests that run longer than
+	// 300s (5 min) generally.
+	// if testing.Short() {
+	// 	t.Skip("skipping long-running test in short mode")
+	// }
```
Comment (YakDriver) [inline internal/service/lexv2models/bot_test.go:30]:
```suggestion
```

File: internal/service/lexv2models/bot_test.go [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+	if acctest.PreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+}
+
+func testAccBotBaseConfig(rName string) string {
```
Comment (YakDriver) [inline internal/service/lexv2models/bot_test.go:210]: Noice

File: website/docs/r/lexv2models_bot.html.markdown [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  idle_session_ttl_in_seconds = 10
+  role_arn                    = "bot_example_arn"
+}
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+* `name` - The name of the bot. The bot name must be unique in the account that creates the bot. Type String. Length Constraints: Minimum length of 1. Maximum length of 100.
```
Comment (YakDriver) [inline website/docs/r/lexv2models_bot.html.markdown:32]:
```suggestion
* `name` - Name of the bot. The bot name must be unique in the account that creates the bot. Type String. Length Constraints: Minimum length of 1. Maximum length of 100.
```

File: website/docs/r/lexv2models_bot.html.markdown [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+* `data_privacy` - Provides information on additional privacy protections Amazon Lex should use with the bot's data. See [`data_privacy`](#data-privacy)
+* `idle_session_ttl_in_seconds` - The time, in seconds, that Amazon Lex should keep information about a user's conversation with the bot. You can specify between 60 (1 minute) and 86,400 (24 hours) seconds.
```
Comment (YakDriver) [inline website/docs/r/lexv2models_bot.html.markdown:34]:
```suggestion
* `idle_session_ttl_in_seconds` - Time, in seconds, that Amazon Lex should keep information about a user's conversation with the bot. You can specify between 60 (1 minute) and 86,400 (24 hours) seconds.
```

Convention (not always followed very well) is to drop "The", "A", and "An" at the beginning of argument descriptions.

File: website/docs/r/lexv2models_bot.html.markdown [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+* `role_arn` - The Amazon Resource Name (ARN) of an IAM role that has permission to access the bot.
```
Comment (YakDriver) [inline website/docs/r/lexv2models_bot.html.markdown:35]:
```suggestion
* `role_arn` - ARN of an IAM role that has permission to access the bot.
```

File: website/docs/r/lexv2models_bot.html.markdown [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+The following arguments are optional:
+
+* `members` - List of bot members in a network to be created. See [`bot_members`](#bot-members).
```
Comment (YakDriver) [inline website/docs/r/lexv2models_bot.html.markdown:39]: This ☝️

File: website/docs/r/lexv2models_bot.html.markdown [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+* `name` (Required) - The unique name of a bot that is a member of this network of bots.
+* `version` (Required) - The version of a bot that is a member of this network of bots.
+
+## Timeouts
+
+[Configuration options](https://developer.hashicorp.com/terraform/language/resources/syntax#operation-timeouts):
+
+* `create` - (Default `60m`)
+* `update` - (Default `180m`)
+* `delete` - (Default `90m`)
```
Comment (YakDriver) [inline website/docs/r/lexv2models_bot.html.markdown:69]:
```suggestion
* `create` - (Default `30m`)
* `update` - (Default `30m`)
* `delete` - (Default `30m`)
```

File: internal/service/lexv2models/bot.go [hunk 11]
```diff
@@ -0,0 +1,584 @@
...
+
+		if err != nil {
+			return nil, "", err
+		}
+
+		return out, aws.ToString((*string)(&out.BotStatus)), nil
+	}
+}
+
+func FindBotByID(ctx context.Context, conn *lexmodelsv2.Client, id string) (*lexmodelsv2.DescribeBotOutput, error) {
```
[the remark this answers is not in the document]
Comment (nam054) [reply internal/service/lexv2models/bot.go:456]: Yes! It's used in the Destroy test

File: internal/service/lexv2models/bot.go (hunk 3 above)
Comment (nam054) [reply internal/service/lexv2models/bot.go:481]: Yes, especially with the intent resource! That one will require a lot of flexing.

**Pull request merged**